services/block_service_test.go

```diff
@@ -123,12 +125,33 @@ func TestBlockService_Online(t *testing.T) {
 	assert.Nil(t, block)
 	assert.Equal(t, ErrKava.Code, err.Code)
 	assert.Equal(t, ErrKava.Message, err.Message)
+	// errors are not retriable
+	assert.Equal(t, ErrKava.Retriable, false)
 	assert.Equal(t, kavaErr.Error(), err.Details["context"])
 
 	blockTransaction, err := servicer.BlockTransaction(ctx, &types.BlockTransactionRequest{})
 	assert.Nil(t, blockTransaction)
 	assert.Equal(t, ErrUnimplemented.Code, err.Code)
 	assert.Equal(t, ErrUnimplemented.Message, err.Message)
 
+	abciErr := tmstate.ErrNoABCIResponsesForHeight{Height: 10001}
+	kavaErr = tmrpctypes.RPCInternalError(tmrpctypes.JSONRPCIntID(1), abciErr).Error
+	mockClient.On(
+		"Block",
+		ctx,
+		blockIdentifier,
+	).Return(
+		nil,
+		kavaErr,
+	).Once()
+
+	block, err = servicer.Block(ctx, &types.BlockRequest{
+		NetworkIdentifier: networkIdentifier,
+		BlockIdentifier:   blockIdentifier,
+	})
+	assert.Nil(t, block)
```

Comment:
Should we be asserting `err` is nil?

Reply:
Good catch -- there was a bug here, fixed and now only modify the error after wrapping